src/TLuaInterpreter.cpp

```diff
@@ -10042,101 +10060,131 @@ int TLuaInterpreter::setGridMode(lua_State* L)
 // Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#setFgColor
 int TLuaInterpreter::setFgColor(lua_State* L)
 {
-    int s = 1;
+    int s = 0;
     int n = lua_gettop(L);
-    string a1;
+    QString windowName;
     int luaRed;
     int luaGreen;
     int luaBlue;
     if (n > 3) {
-        if (lua_isstring(L, s)) {
-            a1 = lua_tostring(L, s);
-            s++;
+        if (!lua_isstring(L, ++s)) {
+            lua_pushfstring(L, "setFgColor: bad argument #%d type (window name expected, got %s!)", s, luaL_typename(L, s));
+            lua_error(L);
```

Comment:
this is where we cheat and say `return lua_error(L);` on one line - it fools some checkers that complain that the return is "unreachable code" ***and*** developers who do not realise that `lua_error(...)` does not return...!

Reply:
applied.